packages/gatsby/src/bootstrap/load-plugins/__tests__/validate.js

```diff
@@ -198,3 +199,25 @@ describe(`handleMultipleReplaceRenderers`, () => {
     expect(result).toMatchSnapshot()
   })
 })
+
+describe(`warnOnIncompatiblePeerDependency`, () => {
+  beforeEach(() => {
+    reporter.warn.mockClear()
+  })
+
+  it(`Does not warn when no peer dependency`, () => {
+    warnOnIncompatiblePeerDependency(`dummy-package`, { peerDependencies: {} })
+
+    expect(reporter.warn.mock.calls.length).toBe(0)
+  })
+
+  it(`Warns on incompatible gatsby peer dependency`, async () => {
+    warnOnIncompatiblePeerDependency(`dummy-package`, {
+      peerDependencies: {
+        gatsby: `<2.0.0`,
+      },
+    })
+
+    expect(reporter.warn.mock.calls.length).toBe(2)
```

Comment:
This is testing an implementation detail effectively, i.e. we _happen_ to be logging twice but there's no guarantee we'll do that in the future.

I'd recommend logging just the once (with new lines) with the approach I show below, and then we can do something like:

```js
expect(reporter.warn).toHaveBeenCalledWith(
  expect.stringContaining(`Plugin dummy-package is incompatible`) // test whatever you'd like here!
)
```